Framework/HistogramData/src/Histogram.cpp

```diff
@@ -140,6 +159,9 @@ void Histogram::setSharedX(const Kernel::cow_ptr<HistogramX> &x) & {
 
   Throws if the size does not match the current size. */
 void Histogram::setSharedY(const Kernel::cow_ptr<HistogramY> &y) & {
+  if (yMode() == YMode::Uninitialized)
+    throw std::logic_error(
+        "Histogram::setSharedY: YMode is not set and cannot be determined");
```

Comment:
@OwenArnold Yes, and there is another reason against putting the flag on `HistogramY`: We introduced that type to have something the client can use when they do not care what the underlying data actually is. If we now put the flag there, we could ask why do we not just have to distinct types and remove `Histogram::y()` altogether.

So I am not sure about this, I will think about it some more.